core/src/host.rs

```diff
@@ -35,6 +36,12 @@ pub trait HostRef {
     fn last_call(&self) -> ContractCallResult;
 }
 
+impl<T: HostRef> Addressable for T {
+    fn address(&self) -> &Address {
+        HostRef::address(self)
```

Comment:
```suggestion
        self.address()
```

Reply:
Must stay as it is. Otherwise, does not know which function is calling (Addressable or HostRef), so must use HostRef explicitly.